### PR TITLE
Fix non-interface type ErrResourceNotFound on left

### PR DESCRIPTION
**Pull request by FengyunPan**
Related to #58145
The gophercloud.ErrResourceNotFound is not a interface, so should
use reflect to get its type then do a check.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by FengyunPan** [comment]
cc @krlng

File: pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go [hunk 1]
```diff
@@ -1002,8 +1003,14 @@ func (lbaas *LbaasV2) ensureSecurityGroup(clusterName string, apiService *v1.Ser
 	lbSecGroupID, err := groups.IDFromName(lbaas.network, lbSecGroupName)
 	if err != nil {
 		// check whether security group does not exist
-		_, ok := err.(*gophercloud.ErrResourceNotFound)
-		if ok {
+		// The security group does not exist while 'err' is object of gophercloud.ErrResourceNotFound
```
Comment (dims) [inline pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go]: maybe we can use a small method? since we are using the same thing 3 times.

Comment (FengyunPan) [reply pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go]: Yeah, good point.

**Pull request merged**